packages/push/src/evaluation/cases/iter.rs

```diff
@@ -0,0 +1,180 @@
+    ///
+    /// ```
+    /// # use push::evaluation::{Case, Cases};
+    /// #
+    /// let cases = Cases::new()
+    ///     .with_case(Case::new("Hello", 5))
+    ///     .with_case(Case::new("People", 6));
+    ///
+    /// let mut iter = (&cases).into_iter();
+    ///
+    /// assert_eq!(iter.next(), Some(&Case::new("Hello", 5)));
+    /// assert_eq!(iter.next(), Some(&Case::new("People", 6)));
+    /// assert_eq!(iter.next(), None);
+    /// ```
+    fn into_iter(self) -> Self::IntoIter {
+        self.cases.iter()
+    }
+}
+
+impl<'a, Input, Output> IntoIterator for &'a mut Cases<Input, Output> {
+    type Item = &'a mut Case<Input, Output>;
+    type IntoIter = std::slice::IterMut<'a, Case<Input, Output>>;
+
+    /// Returns an iterator over mutable references to the cases in the
+    /// `Cases` struct.
+    ///
+    /// This allows the `Cases` struct to be used mutably in a `for` loop or
+    /// other iterator-based operations.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use push::evaluation::{Case, Cases};
+    /// #
+    /// let mut cases = Cases::new()
+    ///     .with_case(Case::new("Hello", 5))
+    ///     .with_case(Case::new("People", 6));
+    ///
+    /// for c in &mut cases {
+    ///     c.output *= 2
+    /// }
+    ///
+    /// let mut iter = cases.into_iter();
+    ///
+    /// assert_eq!(iter.next(), Some(Case::new("Hello", 10)));
+    /// assert_eq!(iter.next(), Some(Case::new("People", 12)));
+    /// assert_eq!(iter.next(), None);
+    /// ```
+    fn into_iter(self) -> Self::IntoIter {
+        self.cases.iter_mut()
+    }
+}
+
+impl<Input, Output, C> FromIterator<C> for Cases<Input, Output>
+where
+    C: Into<Case<Input, Output>>,
+{
+    /// Create a set of cases from an iterator over items that can
+    /// be converted into individual cases.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use push::evaluation::{Case, Cases};
+    /// #
+    /// let items = [("Hello", 5), ("People", 6)];
+    /// let cases = Cases::from_iter(items);
+    ///
+    /// assert!(cases.inputs().eq(&["Hello", "People"]));
+    /// assert!(cases.outputs().eq(&[5, 6]));
+    /// ```
+    fn from_iter<T: IntoIterator<Item = C>>(iter: T) -> Self {
+        Self {
+            cases: iter.into_iter().map(Into::into).collect(),
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::{Case, Cases};
+
+    #[test]
+    fn into_iterator() {
+        let items = [Case::new("Hello", 5), Case::new("you", 3)];
+        let cases = Cases::from_iter(items);
+
+        assert!(cases.into_iter().eq(items));
+    }
+
+    #[test]
+    fn ref_into_iterator() {
+        let cases = Cases::new()
+            .with_case(Case::new("Hello", 5))
+            .with_case(Case::new("People", 6));
+
+        let mut iter = (&cases).into_iter();
+
+        assert_eq!(iter.next(), Some(&Case::new("Hello", 5)));
+        assert_eq!(iter.next(), Some(&Case::new("People", 6)));
+        assert_eq!(iter.next(), None);
```

Comment:
```suggestion
        assert!(iter.eq([&Case::new("Hello",5), &Case::new("People", 6)]);
```

Reply:
Totally! And there are more of these as well. Grrr… I'll make that change and let you know when it's time for a review.